packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,314 @@
+import { Metric, MetricOptions } from '@aws-cdk/aws-cloudwatch';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { CfnCanary } from './synthetics.generated';
+
+/**
+ * Properties for a canary
+ */
+export interface CanaryProps {
+  /**
+   * The s3 location that stores the data of the canary runs.
+   *
+   * @default - A new s3 bucket will be created.
+   */
+  readonly artifactBucket?: s3.IBucket;
+
+  /**
+   * Canary execution role.
+   *
+   * This is the role that will be assumed by the canary upon execution.
+   * It controls the permissions that the canary will have. The Role must
+   * be assumable by the 'lambda.amazonaws.com' service principal.
+   *
+   * The default Role automatically has permissions granted for Canary execution.
+   * If you provide a Role, you must add the relevant policies.
+   *
+   * The relevant policies are "s3:PutObject", "s3:GetBucketLocation", "s3:ListAllMyBuckets",
+   * "cloudwatch:PutMetricData", "logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents".
+   * @see required permissions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-synthetics-canary.html
+   *
+   * @default - A unique role will be generated for this canary.
+   * Both supplied and generated roles can always be changed by calling 'addToRolePolicy'.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * How many seconds the canary should run before timing out. Minimum time is 60 seconds and maximum is 900 seconds.
+   *
+   * @default - the minimum between frequency and 900 seconds
+   */
+  readonly timeout?: cdk.Duration;
```

Comment:
I think we should remove this property from the API for now, it's use case in unclear and it seems the properties in `schedule` are sufficient. We can introduce it at a later stage when we will learn the usage pattern, if any exists.

Reply:
okay -- I will hardcode the default of min(frequency, 900). I agree that the use case is unclear at this point.

I will say that this property is leaked from lambda where lambda's `timeout` is useful because the execution time affects cost.

While canaries are priced by runs, "For each canary run, you may incur additional charges for other AWS services utilized such as AWS Lambda, Amazon S3, and CloudWatch Logs." ([source](https://aws.amazon.com/cloudwatch/pricing/) -- in the 'canaries' section of the Paid Tier). 

It might be a leaky abstraction, it might be a necessary cost measure. I will remove for now.

Reply:
> 'I will hardcode the default of min(frequency, 900)'

Why do you need to hardcode? I think this is optional in the latest spec

Reply:
Here's my thought process: `runConfig` is now optional, but at least according the the specs, _if specified_, it requires the `timeoutInSeconds` property. So for CFN, the user cannot specify `memorySize` without specifying `timeoutInSeconds`. That's why I feel I need to hardcode in the `timeout` -- if we remove it entirely, the user can no longer specify `memorySize`. 

Alternatively, I could only add in the hardcoded `timeout` _if_ `memorySize` is specified, but I feel like this is needlessly complicated.

Reply:
Here is the code in synthetics.generated:

```ts
    export interface RunConfigProperty {
        /**
         * `CfnCanary.RunConfigProperty.MemoryInMB`
         * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-memoryinmb
         */
        readonly memoryInMb?: number;
        /**
         * `CfnCanary.RunConfigProperty.TimeoutInSeconds`
         * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-timeoutinseconds
         */
        readonly timeoutInSeconds: number;
    }
```

As you can see, while `runConfig` is optional, if specified it requires `timeoutInSeconds`. It is very weird.